.github/workflows/cpp-python-tests.yml

```diff
@@ -4,105 +4,129 @@ name: cpp-python-tests
+        run: |
+          echo "::set-output name=release_file_name::opentera-webrtc-native-client-${{runner.os}}-\
+          ${{ steps.generate_os_release_name_output.outputs.os_release_name }}-\
+          ${{ steps.generate_os_release_version_output.outputs.os_release_version }}-\
+          ${{ steps.generate_os_release_architecture_output.outputs.os_release_architecture }}-\
+          ${{ steps.generate_version_output.outputs.version }}"
+
+      - name: Print Release File Name
+        run: echo "Release File Name ${{ steps.generate_release_file_name.outputs.release_file_name }}"
+        shell: bash
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build cmake build-essential libssl-dev libboost-all-dev
+          sudo apt-get install libglib2.0-dev libgtk-3-dev libpulse-dev libasound2-dev tree
+
+      - name: Install python dependencies
+        run: |
+          python3 -m pip install --user --upgrade pip
+          python3 -m pip install --user --upgrade setuptools wheel
+          python3 -m pip install --user --upgrade twine
+          python3 -m pip install --user --upgrade pytest
+
+      - name: Create build directory
+        working-directory: opentera-webrtc-native-client
+        run: |
+          mkdir build
+
+      - name: Run CMake and build opentera-webrtc-native-client
+        working-directory: opentera-webrtc-native-client/build
+        run: |
+          cmake ../ -DCMAKE_BUILD_TYPE=${{ matrix.build-type }}
+          make -j2
+          make install
+
+      - name: Python installation cleanup
+        working-directory: opentera-webrtc-native-client/build/OpenteraWebrtcNativeClient/python/package/opentera_webrtc_native_client
+        run: |
+          rm -rfv build
+          rm -rfv '*.so'
+          rm -rfv opentera_webrtc_native_client.egg-info
+          rm -rfv opentera_webrtc_native_client.html
+
+      - name: Install python dependencies for signaling-server
+        working-directory: signaling-server
+        run: |
+          if [ -f requirements.txt ]; then pip3 install -r requirements.txt; fi
+
+      - name: Run CPP Tests
+        working-directory: opentera-webrtc-native-client/build/bin
+        run: |
+          ./OpenteraWebrtcNativeClientTests
+
+      - name: Run Python signaling-server Tests
+        working-directory: signaling-server
+        run: |
+          python3 -m pytest -v ${{ pytest_stdout.outputs.value }} opentera/webrtc/tests
+
+      - name: Run Python webrtc-native-client Tests
+        working-directory: opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/test
+        run: |
+          if [ -f requirements.txt ]; then pip3 install -r requirements.txt; fi
+          cp ../../../build/OpenteraWebrtcNativeClient/python/bin/${{ matrix.build-type }}/_opentera_webrtc_native_client.cpython*.so .
+          python3 -m pytest -v ${{ pytest_stdout.outputs.value }}
```

Comment:
Replaced unittest with pytest as the output is cleaner, it allows more customization, and it shows warnings